k8-operator/controllers/infisicalsecret_helper.go

```diff
@@ -28,19 +58,102 @@ const OPERATOR_SETTINGS_CONFIGMAP_NAME = "infisical-config"
 const OPERATOR_SETTINGS_CONFIGMAP_NAMESPACE = "infisical-operator-system"
 const INFISICAL_DOMAIN = "https://app.infisical.com/api"
 
-type AuthStrategyType string
+var infisicalClient = infisicalSdk.NewInfisicalClient(infisicalSdk.Config{})
```

Comment:
This might cause issues with concurrency when many instances of InfisicalSecrets are running in the same K8s cluster managed by the same operator. It might also make it difficult to change the configuration of the SDK based on different InfisicalSecret contexts. I suggest a better place to instantiate the SDK is in the main `Reconcile` function in the `infisicalsecret_controller.go` file. Since that is the entry point where all of this code gets triggered, you can parse InfisicalSecret there and create an instance of the SDK with its siteUrl and other configurations.

Example:

```
func (r *InfisicalSecretReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
...
	// Initialize the SDK client with the necessary configuration
	infisicalClient := infisicalSdk.NewInfisicalClient(infisicalSdk.Config{
		SiteUrl:   api.API_HOST_URL,
		UserAgent: api.USER_AGENT_NAME,
	})

	err = r.ReconcileInfisicalSecret(ctx, infisicalSecretCR, infisicalClient)
....
}
```